amethyst_input/src/bindings.rs

```diff
@@ -9,7 +9,8 @@ use smallvec::SmallVec;
 use super::{Axis, Button};
 
 /// Used for saving and loading input settings.
-#[derive(Serialize, Deserialize, Clone)]
+#[derive(Derivative, Serialize, Deserialize, Clone)]
+#[derivative(Default(bound = "AX: Hash + Eq, AC: Hash + Eq"))]
```

Comment:
Why does derive not work here?

Reply:
Because `fn new()` is implemented for those bounds.

Reply:
Because otherwise we'd have different semantics between Default and `fn new()` depending on bounds. I don't know if we want to unrestrict `fn new()` or not?

Reply:
I usually don't put bounds on the new function except for when I don't want to create another impl block.